### ISSUE TITLE
Improve Navigator and Faceted Search tracking mechanism 

### ISSUE THREAD

**Issue by cyberandy** [insideout10/wordlift-plugin#1226]
Please describe the issue here:

> UTM parameters are intended to be used in external campaigns (for example, promoted posts on Facebook). Unfortunately, they are not suitable for tracking internal campaigns. 

In order not to corrupt Google Analytics data when using UTM parameters for the tracking of our widget we shall define a new tracking method.

As we do this we shall pave the way to gather the data that we need to improve the content recommendations engine that WordLift uses. In order to do so we should track the following data points:

- `Timestamp`—the timestamp of the event. The click events we are interested in 
- `User ID`—the unique ID of the user, or the unique client ID. Assuming here we're working only for users that have GA we can use the cookie that stores the unique identifier for the browser client used in the web session
- `Article ID`—the unique ID of the article. This could be the URI of the Article in our datastore so that we can keep data also when the URL is updated (and we know this can happen)
- `Related entities`—the list of entities the article has been annotated with.
- `Time On Page`—the amount of time in seconds that the user spent viewing the article. Time On Page is not explicitly tracked in page tracking events for Google Analytics. We can calculate the the time on page by subtracting the time of the page tracking event in which the user accessed the page from the time of the next page tracking event for that user.

**Comment by ziodave** [comment]
![](https://github.trello.services/images/mini-trello-icon.png) [Track Navigator and Faceted Search clicks *not* using UTM](https://trello.com/c/E4ArSDoN/20-track-navigator-and-faceted-search-clicks-not-using-utm)

**Comment by akshayraje** [comment]
Hi @cyberandy, when I started implementation of this, I realised that we may need to re-look at the spec for this. The assumption here is that we fire the GA event based on the GA object present on the page (i.e. gtag, dataLayer (gtm), ga or __gaTracker). Also now that we lazy load the widgets, we have the flexibility to fire the event when it is actually seen by the user. Here are my notes / observations on this:

- `Timestamp` - All GA libraries send this by default and it is not recommended that we explicitly send a timestamp with event so I suggest we drop this.
- `User ID` - In most cases we would not have a user context (of a logged in user). The unique client ID (an pseudonymous device or browser instance) is already [managed by all GA libraries](https://support.google.com/analytics/answer/6205850?hl=en) and hence we should not manually generate / send this with the event.
- `Article ID` - Agree on this. No change proposed.
- `Related entities` - Agree on this. No change proposed.
- `Time On Page` - Tracking time on page as mentioned above is not feasible now as the event itself will only be fired when the widget comes in the active viewport.

To summarise, here's what I propose we send in the GA event:
- `Widget name` - Navigator, Products Navigator, Faceted Search or Context Card
- `Article ID` - Unique ID of the article- the URI of the Article in our datastore
- `Related entities` - List of entities the article has been annotated with.

Let me know if this works so that I can move ahead with implementation.
